xtask/src/lib.rs

```diff
@@ -146,6 +146,11 @@ pub fn build_documentation(
         builder = builder.arg("--open");
     }
 
+    // If targeting an Xtensa device, we must use the '+esp' toolchain modifier:
+    if target.starts_with("xtensa") {
```

Comment:
In the back of my head I'm wondering if this will cause issues down the line, but I don't have any real argument not to do this right now, I suppose we could always remove it if it does.

Reply:
I can't really think of a situation where this would be problematic, we're in control of the Xtensa toolchain and this repository, so any changes made will be made by us. I think it's fine (at least for now).